src/integrators/controllers.jl

```diff
@@ -736,3 +738,81 @@ function step_accept_controller!(integrator, alg::Union{FBDF{max_order}, DFBDF{m
     integrator.cache.iters_from_event += 1
     return integrator.dt / q
 end
+
+
+
+# Relaxation step size controller
+"""
+    RelaxationController(controller, T)
+
+Controller to perform a relaxation on a step of a Runge-Kuttas method.  
+
+## References
+ - Sebastian Bleecke, Hendrik Ranocha (2023)
+    Step size control for explicit relaxation Runge-Kutta methods preserving invariants
+    [DOI: 10.1145/641876.641877](https://doi.org/10.1145/641876.641877)
+"""
+mutable struct RelaxationController{CON, T} <: AbstractController
+    controller::CON
+    gamma::T
+    function RelaxationController(controller::AbstractController, T)
+        new{typeof(controller), T}(controller, one(T))
+    end
+end
+
+mutable struct Relaxation{INV, OPT, T}
+    invariant::INV
+    opt::OPT
+    gamma_min::T
+    gamma_max::T
+    function Relaxation(invariant, opt = AlefeldPotraShi, gamma_min = 4//5, gamma_max = 6//5)
```

Comment:
What is `AlefeldPotraShi`? It doesn't seem to be defined anywhere?

Reply:
never mind I see it's the alg.

Reply:
ITP is likely a better choice here.

Reply:
yeah my version of this pr switched to ITP and removes the Roots dependency